Sources/Datadog/URLSessionAutoInstrumentation/DDURLSessionDelegate.swift

```diff
@@ -67,22 +91,27 @@ open class DDURLSessionDelegate: NSObject, URLSessionTaskDelegate, URLSessionDat
     /// passed to `DatadogConfiguration.Builder` via `trackURLSession(firstPartyHosts:)`
     @objc
     public convenience init(additionalFirstPartyHosts: Set<String>) {
-        self.init(additionalFirstPartyHostsWithHeaderTypes: additionalFirstPartyHosts.reduce(into: [:], { partialResult, host in
-            partialResult[host] = [.datadog]
-        }))
+        self.init(
+            in: nil,
+            additionalFirstPartyHostsWithHeaderTypes: additionalFirstPartyHosts.reduce(into: [:], { partialResult, host in
+                partialResult[host] = [.datadog]
+            })
+        )
     }
 
     /// Automatically tracked hosts can be customized per instance with this initializer.
     ///
+    /// **NOTE:** If `trackURLSession(firstPartyHostsWithHeaderTypes:)` is never called, automatic tracking will **not** take place.
+    ///
     /// - Parameters:
-    ///   - core: Datadog SDK core.
-    ///   - additionalFirstPartyHosts: additionalFirstPartyHosts: these hosts are tracked **in addition to** what was
+    ///   - core: Datadog SDK instance (or `nil` to use default SDK instance).
+    ///   - additionalFirstPartyHosts: these hosts are tracked **in addition to** what was
     ///                                passed to `DatadogConfiguration.Builder` via `trackURLSession(firstPartyHosts:)`
     public init(
-        in core: DatadogCoreProtocol,
-        additionalFirstPartyHostsWithHeaderTypes: [String: Set<TracingHeaderType>]
+        in core: DatadogCoreProtocol? = nil,
+        additionalFirstPartyHostsWithHeaderTypes: [String: Set<TracingHeaderType>] = [:]
```

Comment:
This change in public API is backward compatible, even if someone is using `defaultDatadogCore` (for whatever reason - as we consider it "publicly internal"):
```swift
// still works after this change:
DDURLSessionDelegate(in: defaultDatadogCore, additionalFirstPartyHostsWithHeaderTypes: [:])
```